src/ldp/auxiliary/AuxiliaryStrategy.ts

```diff
@@ -0,0 +1,42 @@
+import type { Representation } from '../representation/Representation';
+import type { RepresentationMetadata } from '../representation/RepresentationMetadata';
+import type { ResourceIdentifier } from '../representation/ResourceIdentifier';
+import type { AuxiliaryIdentifierStrategy } from './AuxiliaryIdentifierStrategy';
+
+/**
+ * A strategy for handling one or more types of auxiliary resources.
+ * References to "an auxiliary resource" implicitly imply a specific type of auxiliary resources
+ * supported by this strategy.
+ */
+export interface AuxiliaryStrategy extends AuxiliaryIdentifierStrategy {
+  /**
+   * Whether this auxiliary resource can be deleted when it's in a root storage container.
+   * @param identifier - Identifier of the auxiliary resource.
+   */
+  canDeleteRoot: (identifier: ResourceIdentifier) => boolean;
```

Comment:
Also, can this be just a getter? Not a function?

Reply:
Then we can no longer pass the ResourceIdentifier parameter along, which is required for the composite strategy.

Reply:
Woops, of course. Would still look into naming though, especially with the param.